src/daft-core/src/array/ops/utf8.rs

```diff
@@ -260,6 +260,81 @@ impl Utf8Array {
         Ok(Utf8Array::from((self.name(), Box::new(arrow_result))))
     }
 
+    pub fn left(&self, n: &UInt32Array) -> DaftResult<Utf8Array> {
```

Comment:
You can take advantage of our `DaftIntegerType` trait https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/datatypes/mod.rs#L305, to make this a generic function. example usage here: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/sort.rs#L67-L70